packages/lesswrong/server/scripts/mergeAccounts.ts

```diff
@@ -451,20 +452,15 @@ async function recomputeKarma(userId: string) {
   const selector: Record<string, any> = {
     authorIds: user._id,
     userId: {$ne: user._id},
-    cancelled: false
+    cancelled: false,
+    collectionName: {$in: collectionsThatAffectKarma}
   };
-  if (Votes.isPostgres()) {
-    selector["legacyData.legacy"] = {$ne: true};
-  } else {
-    selector.legacy = {$ne: true};
-  }
   const allTargetVotes = await Votes.find(selector).fetch()
-  const totalNonLegacyKarma = sumBy(allTargetVotes, vote => {
-    return vote.power
+  const karma = sumBy(allTargetVotes, vote => {
+    // a doc author cannot give karma to themselves or any other authors for that doc
+    return vote.authorIds.includes(vote.userId) ? 0 : vote.power
   })
-  // @ts-ignore FIXME legacyKarma isn't in the schema, figure out whether it's real
-  const totalKarma = totalNonLegacyKarma + (user.legacyKarma || 0)
-  return totalKarma
+  return karma
```

Comment:
According to JP, we're no longer doing something different with legacy data, so now we can treat legacy votes as normal votes.